test/libsolidity/syntaxTests/functionTypes/conversion_to_address.sol

```diff
@@ -0,0 +1,7 @@
+contract C {
+    function f() public view returns (address) {
+        return address(this.f);
+    }
+}
+// ----
+// TypeError: (77-92): Explicit type conversion not allowed from "function () view external returns (address)" to "address".
```

Comment:
Can you add something to `TypeChecker::typeCheckTypeConversionAndRetrieveReturnType` to make the error message suggest how to fix it?

Reply:
I can have a look.

Reply:
I put it in a secondary source location, so it won't occur in the test expectations.